sql/plan/loop.go

```diff
@@ -125,6 +167,36 @@ func (l *loopIter) Next(ctx *sql.Context) (sql.Row, error) {
 	const maxIterationCount = 10_000_000_000
 	l.loopIteration++
 	for ; l.loopIteration < maxIterationCount; l.loopIteration++ {
+		// If the condition is false, then we stop evaluation
+		condition, err := l.condition.Eval(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+		conditionBool, err := sql.ConvertToBool(condition)
+		if err != nil {
+			return nil, err
+		}
+		if !conditionBool {
+			return nil, io.EOF
+		}
+
+		if l.blockIter == nil {
+			var err error
+			l.blockIter, err = l.block.RowIter(ctx, nil)
+			if controlFlow, ok := err.(loopError); ok && strings.ToLower(controlFlow.Label) == strings.ToLower(l.label) {
+				if controlFlow.IsExit {
+					return nil, io.EOF
+				} else {
+					err = io.EOF
+				}
+			}
+			if err == io.EOF {
+				l.blockIter = sql.RowsToRowIter()
+			} else if err != nil {
+				return nil, err
+			}
```

Comment:
(minor) you might consider extracting this into a private function on `Loop` to get the iterator. Might make this block and the block on lines 114-126 a little easier to read.